File: app/src/main/java/com/p4r4d0x/genreclassifier/rest/ClassifyRetrofitService.java
package com.p4r4d0x.genreclassifier.rest;

import com.p4r4d0x.genreclassifier.rest.classify.ClassifyRequest;
import com.p4r4d0x.genreclassifier.rest.classify.ClassifyResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface ClassifyRetrofitService {

    @POST("dev/classifier/classifygenre")
    Call<ClassifyResponse> classifySong(@Body ClassifyRequest classifyRequest);


}

File: app/src/main/java/com/p4r4d0x/edmgenreclassifier/datasource/retrofit/ClassifyRetrofitService.kt
package com.p4r4d0x.edmgenreclassifier.datasource.retrofit

import com.p4r4d0x.edmgenreclassifier.datasource.dto.GenreClassifyRequestDTO
import com.p4r4d0x.edmgenreclassifier.datasource.dto.GenreClassifyResponseDTO
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface ClassifyRetrofitService {
    @POST("dev/classifier/classifygenre")
    fun classifySong(@Body classifyRequest: GenreClassifyRequestDTO?): Call<GenreClassifyResponseDTO?>?
}